pkg/publish/default.go

```diff
@@ -203,7 +208,13 @@ func (d *defalt) Publish(ctx context.Context, br build.Result, s string) (name.R
 	// https://github.com/google/go-containerregistry/issues/212
 	s = strings.ToLower(s)
 
-	ro := []remote.Option{remote.WithAuth(d.auth), remote.WithTransport(d.t), remote.WithContext(ctx), remote.WithUserAgent(d.userAgent)}
+	ro := []remote.Option{remote.WithTransport(d.t), remote.WithContext(ctx), remote.WithUserAgent(d.userAgent)}
+	if d.auth != authn.Anonymous {
```

Comment:
This should just always pass `remote.WithAuthFromKeychain(d.keychain)`, which is `staticKeychain{authn.Anonymous}` by default.